skiko/src/awtMain/objectiveC/macos/MetalContextHandler.mm

```diff
@@ -0,0 +1,64 @@
+#ifdef SK_METAL
+
+#import <jawt.h>
+#import <jawt_md.h>
+
+#import <QuartzCore/CAMetalLayer.h>
+#import <Metal/Metal.h>
+#import <GrDirectContext.h>
+#import <mtl/GrMtlBackendContext.h>
+#import <mtl/GrMtlTypes.h>
+
+#import "MetalDevice.h"
+
+extern "C"
+{
+JNIEXPORT jlong JNICALL Java_org_jetbrains_skiko_context_MetalContextHandler_makeMetalContext(
+    JNIEnv* env, jobject contextHandler, jlong devicePtr)
+{
+    @autoreleasepool {
+        MetalDevice *device = (__bridge MetalDevice *) (void*) devicePtr;
+        GrMtlBackendContext backendContext = {};
+        backendContext.fDevice.retain((__bridge GrMTLHandle) device.adapter);
+        backendContext.fQueue.retain((__bridge GrMTLHandle) device.queue);
+        return (jlong) GrDirectContext::MakeMetal(backendContext).release();
+    }
+}
+
+JNIEXPORT jlong JNICALL Java_org_jetbrains_skiko_context_MetalContextHandler_makeMetalRenderTarget(
+    JNIEnv * env, jobject contextHandler, jlong devicePtr, jint width, jint height)
+{
+    @autoreleasepool {
+        MetalDevice *device = (__bridge MetalDevice *) (void *) devicePtr;
+        GrBackendRenderTarget* renderTarget = NULL;
+
+        id<CAMetalDrawable> currentDrawable = [device.layer nextDrawable];
+        if (!currentDrawable) return 0;
+        device.drawableHandle = currentDrawable;
+        GrMtlTextureInfo info;
+        info.fTexture.retain((__bridge GrMTLHandle) currentDrawable.texture);
+        renderTarget = new GrBackendRenderTarget(width, height, 0, info);
```

Comment:
I do not see deleting (freeing memory) of this object

Reply:
I haven't changed how code works. This PR is only about refactorings, if you think that memory leak can be here, I would be glad if you fix it in the separate PR

Reply:
As I understand we provide this reference to jvm and there is `renderTarget = null` that will clear this reference. Doesn't it work?

Reply:
No, operator `new` MUST be paired with `delete` (it might be not explicit in case of using smart pointer classes from std lib, but it's not a case here.
For not it looks like a leak

Reply:
I think this `delete` is here https://github.com/JetBrains/skiko/blob/60ccb70a4cf8b7db8a1dbeaf7a9ecd905d27a6cb/skiko/src/jvmMain/cpp/common/BackendRenderTarget.cc#L7

skiko has quite complicated management of native resources though.
This particular piece is managed by `class BackendRenderTarget(...) : Managed(ptr, _FinalizerHolder.PTR)`

Reply:
Ok, the type is the same, but are we sure that this particular object will go through that path? Shouldn't we declare another finalizer for MetalContextHandler?